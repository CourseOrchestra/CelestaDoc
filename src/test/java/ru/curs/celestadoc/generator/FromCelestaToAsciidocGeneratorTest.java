package ru.curs.celestadoc.generator;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.stream.Collectors;
import java.util.stream.Stream;

class FromCelestaToAsciidocGeneratorTest {

    @Test
    void test() throws Exception {
        final Path adoc = Files.createTempFile("cdtest", ".adoc");
        final Path pu = Paths.get(adoc.toString().replace(".adoc", ".pu"));
        System.out.println(adoc);
        System.out.println(pu);
        try {
            try (FromCelestaToAsciidocGenerator generator = new FromCelestaToAsciidocGenerator(
                    "src/test/resources",
                    adoc.toString(), "_",
                    true, "")) {
                generator.generate();
            }

            final String files = Stream.of(
                    Files.lines(adoc),
                    Stream.of("--------------"),
                    Files.lines(pu))
                    .reduce(Stream::concat)
                    .orElseGet(Stream::empty).collect(Collectors.joining("\n"));
            Approvals.verify(files);
        } finally {
            Files.delete(adoc);
            Files.delete(pu);
        }
    }

}