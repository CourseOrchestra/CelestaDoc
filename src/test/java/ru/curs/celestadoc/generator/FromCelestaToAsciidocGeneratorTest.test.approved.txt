= Database structure report
Automatically generated by Celesta
:!linkcss:
:encoding: UTF-8
:scheme: Scheme:
:table: Table:
:table-in-th: Table
:field: Field
:key: Key field
:specification: Specification
:description: Description
:key-icon: {nbsp}(key)
:foreign-keys: Foreign keys
:is-referenced: Is referenced by
:openq: "
:closeq: "
:fkey-cols-spec: 1,1
:field-cols-specs: 1,1,2

//tag::generated_from_celesta_sql[]
[discrete]
== {scheme} {openq}security{closeq}
//tag::celestareporter_s_security[]

    
[[celestareporter_t_security_email]]
[discrete]
=== {table} {openq}email{closeq}
//tag::celestareporter_t_security_email[]

    

[options="header", cols="{field-cols-specs}"]
|====
^.^|{field} ^.^|{specification} ^.^|{description}
    

|email{key-icon} |VARCHAR(255) NOT NULL |
    

|verification_code |VARCHAR(16) NOT NULL |
    

|code_sent |DATETIME |
    

|verified |BIT NOT NULL DEFAULT 'FALSE' |
    

|attempts_left |INT NOT NULL DEFAULT 5 |
    

|====
    

//tag::is_referenced[]
[discrete]
==== {is-referenced}
    

* <<celestareporter_t_security_user,security.user>>
    

//end::is_referenced[]
    

//end::celestareporter_t_security_email[]
    
[[celestareporter_t_security_user]]
[discrete]
=== {table} {openq}user{closeq}
//tag::celestareporter_t_security_user[]

    

[options="header", cols="{field-cols-specs}"]
|====
^.^|{field} ^.^|{specification} ^.^|{description}
    

|username{key-icon} |VARCHAR(255) NOT NULL |
    

|password |VARCHAR(127) NOT NULL |
    

|is_active |BIT NOT NULL DEFAULT 'TRUE' |
    

|email |VARCHAR(255) |
    

|admin |BIT NOT NULL DEFAULT 'FALSE' |
    

|operator |BIT NOT NULL DEFAULT 'FALSE' |
    

|====
    

//tag::is_referenced[]
[discrete]
==== {is-referenced}
    

* <<celestareporter_t_security_SecurityLog,security.SecurityLog>>
    

//end::is_referenced[]
    

//tag::references[]
[discrete]
==== {foreign-keys}
[options="header", cols="{fkey-cols-spec}"]
|====
^.^|{key} ^.^|{table-in-th}
    

| *email*:email


|<<celestareporter_t_security_email, security.email>>
    
|====
//end::references[]
    

//end::celestareporter_t_security_user[]
    
[[celestareporter_t_security_SecurityLog]]
[discrete]
=== {table} {openq}SecurityLog{closeq}
//tag::celestareporter_t_security_SecurityLog[]

    

[options="header", cols="{field-cols-specs}"]
|====
^.^|{field} ^.^|{specification} ^.^|{description}
    

|id{key-icon} |INT NOT NULL |
    

|username |VARCHAR(255) NOT NULL |
    

|timestamp |DATETIME NOT NULL DEFAULT GETDATE() |
    

|description |VARCHAR(255) NOT NULL |
    

|entity_type |VARCHAR(255) NOT NULL |
    

|entity_id |INT NOT NULL |
    

|affiliate_id |INT |
    

|duration_ms |INT NOT NULL |
    

|====
    

//tag::references[]
[discrete]
==== {foreign-keys}
[options="header", cols="{fkey-cols-spec}"]
|====
^.^|{key} ^.^|{table-in-th}
    

| *username*:username


|<<celestareporter_t_security_user, security.user>>
    
|====
//end::references[]
    

//end::celestareporter_t_security_SecurityLog[]
    

//end::celestareporter_s_security[]
//end::generated_from_celesta_sql[]
    

//tag::generated_from_celesta_sql[]
[discrete]
== {scheme} {openq}foo{closeq}
//tag::celestareporter_s_foo[]

    
[[celestareporter_t_foo_plan]]
[discrete]
=== {table} {openq}plan{closeq}
//tag::celestareporter_t_foo_plan[]

    

[options="header", cols="{field-cols-specs}"]
|====
^.^|{field} ^.^|{specification} ^.^|{description}
    

|id{key-icon} |INT NOT NULL |
    

|point |VARCHAR(30) |
    

|====
    

//end::celestareporter_t_foo_plan[]
    

//end::celestareporter_s_foo[]
//end::generated_from_celesta_sql[]
    
--------------
@startuml
!include 
class email [[celestareporter_t_security_email]] {
- email
verification_code
code_sent
verified
attempts_left
}
class user [[celestareporter_t_security_user]] {
- username
password
is_active
email
admin
operator
}
user "*" -- "1" email
class SecurityLog [[celestareporter_t_security_SecurityLog]] {
- id
username
timestamp
description
entity_type
entity_id
affiliate_id
duration_ms
}
SecurityLog "*" -- "1" user
@enduml
class plan [[celestareporter_t_foo_plan]] {
- id
point
}
@enduml